Bitbucket connector

chandru094 opened the pull request.

huilinn commented:
Auth doesn't seem to be configured properly:
<img width="1280" alt="screen shot 2016-08-13 at 12 14 52 am" src="https://cloud.githubusercontent.com/assets/6780917/17642049/1a30aa8e-60ed-11e6-8a0c-2a92e27beac7.png">

chandru094 commented:
Hi

Can you please check if the callback URL for the consumer you created in
Butbucket is correct.

Regards,
Chandra Prakash R

On Sat, Aug 13, 2016 at 1:00 PM, Yang Huilin <notifications@github.com>
wrote:

> Auth doesn't seem to be configured properly:
> [image: screen shot 2016-08-13 at 12 14 52 am]
> <https://cloud.githubusercontent.com/assets/6780917/17642049/1a30aa8e-60ed-11e6-8a0c-2a92e27beac7.png>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/workato/connector_sdk/pull/33#issuecomment-239607733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARLAJ69XgYqWFulJ9Jo1HyBs4tU4R03Aks5qfXKWgaJpZM4Ji0H5>
> .
>

huilinn commented:
Oh, @chandru094 I think I was using your localhost client ID and secret. When I replaced it with your production ID and secret in the code, it worked.

However, I connected immediately to my browser's Bitbucket account with no regards to what I type in for the input fields - does this mean we don't actually need the input fields to be there?
[![https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c](https://i.gyazo.com/f276e5e1cff9c6fc704838d126b6071c.gif)](https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c)

chandru094 commented:
The username and repository name are required in all the endpoints url. So
I added them in the connection.

On Saturday 13 August 2016, Yang Huilin <notifications@github.com> wrote:

> Oh, @chandru094 <https://github.com/chandru094> I think I was using your
> localhost client ID and secret. When I replaced it with your production ID
> and secret in the code, it worked.
>
> However, I connected immediately to my browser's Bitbucket account with no
> regards to what I type in for the input fields - does this mean we don't
> actually need the input fields to be there?
> [image: https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c]
> <https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/workato/connector_sdk/pull/33#issuecomment-239613844>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARLAJ7-qbCz9ln76WEZXYowdZ8rkoR5Pks5qfZllgaJpZM4Ji0H5>
> .
>

huilinn commented:
Ah okay I see. By the way @chandru094 can you change the client id and secret in the PR? Right now they're still configured to your localhost creds. Production ones:

       client_id: 'HV8cTJ8aLyKC8jucUh',
       client_secret: 'BGHemmMqvXTkPAdCryzTw5GWcw82M8Vx',

chandru094 commented:
I have changed the client id and client secret now to be used by production accounts.

huilinn commented:
Actually, @chandru094, it looks like what's required to be passed into every API call is not username and repo slug, but owner's username and repo slug. So the hint should be changed for the connection to say that the username passed in must be that of the owner of the repo.

Was looking at another API call:
<img width="819" alt="screen shot 2016-08-14 at 1 37 59 am" src="https://cloud.githubusercontent.com/assets/6780917/17648118/c1c1cdc8-61bf-11e6-8307-2d712f577002.png">

huilinn commented:
Also, I think we're currently not handling the case for repos with >1 word. The slug would be all lowercase, with - replace spaces. For example, my site's repo name and slug name:
<img width="1280" alt="screen shot 2016-08-14 at 1 58 09 am" src="https://cloud.githubusercontent.com/assets/6780917/17648243/ac0865ac-61c2-11e6-9536-8c6535218008.png">

The API calls we're making simply passes the user's entered repo name into the call currently, we should change that.

huilinn commented:
Instead of santhosh requesting permission, we should have Workato requesting permission, for a global scope app: <img width="1280" alt="screen shot 2016-08-14 at 2 06 05 am" src="https://cloud.githubusercontent.com/assets/6780917/17648282/ba17761e-61c3-11e6-9cef-170aefd26842.png">

huilinn commented:
Also, for priority and kinds fields, it would be better if they were freeform text fields instead of picklists (we can also have them defaulted to text fields, and then toggle to picklists), because we want that flexibility to assign different priorities and kinds when creating an issue. If it was a picklist, we would be hardcoding all issues to be a specific priority and specific kind, and that's not good for a recipe.

Generally, most fields shouldn't be a picklist field (i.e. issue state, issue type, issue priority). 

<img width="1063" alt="screen shot 2016-08-14 at 2 13 57 am" src="https://cloud.githubusercontent.com/assets/6780917/17648323/c9cafd00-61c4-11e6-810a-1c3a6ec09f7d.png">

chandru094 commented:
If we pass any other values other than values given in the api it throws an error saying not a valid value.

eeshan92 commented:
@chandru094 @huilinn toggle feature will fix that problem. In the meantime, let's change it to a free-text field for the use case

eeshan92 commented on oauth2/bitbucket_connector.rb:
```diff
@@ -0,0 +1,266 @@
+{
+  title: 'bitbucket',
```
Titleize this to `Bitbucket`

eeshan92 commented on oauth2/bitbucket_connector.rb:
```diff
@@ -0,0 +1,266 @@
+      client_secret: 'BGHemmMqvXTkPAdCryzTw5GWcw82M8Vx',
+
+      credentials: ->(connection, access_token) {
+      headers('Authorization': "Bearer #{access_token}")
```
should be indented
